decidim-comments/app/commands/decidim/comments/vote_comment.rb

```diff
@@ -25,26 +25,51 @@ def initialize(comment, author, options = { weight: 1 })
       def call
         case @weight
         when 1
-          vote = @comment.up_votes.find_by(author: @author)
-          if vote
-            vote.destroy!
+          previous_vote = @comment.up_votes.find_by(author: @author)
+          if previous_vote
+            previous_vote.destroy!
           else
-            @comment.up_votes.create!(author: @author)
+            @vote = @comment.up_votes.create!(author: @author)
           end
         when -1
-          vote = @comment.down_votes.find_by(author: @author)
-          if vote
-            vote.destroy!
+          previous_vote = @comment.down_votes.find_by(author: @author)
+          if previous_vote
+            previous_vote.destroy!
           else
-            @comment.down_votes.create!(author: @author)
+            @vote = @comment.down_votes.create!(author: @author)
           end
         else
           return broadcast(:invalid)
         end
+
+        notify_comment_author if @vote
         broadcast(:ok, @comment)
       rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotUnique
         broadcast(:invalid)
       end
+
+      def notify_comment_author
+        Decidim::EventsManager.publish(
+          event: "decidim.events.comments.comment_#{upvote? ? "upvoted" : "downvoted"}",
+          event_class: upvote? ? Decidim::Comments::CommentUpvotedEvent : Decidim::Comments::CommentDownvotedEvent,
+          resource: @comment.commentable,
+          affected_users: [@author],
```

Comment:
Just curious, do we really need to save the vote author here? Isn't

Reply:
Thanks for the heads-up! I was notifying the author of the vote as well!